src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/transfercase/TransferCaseStayedService.java

```diff
@@ -0,0 +1,150 @@
+import uk.gov.hmcts.cmc.claimstore.config.CaseStayedIncrementConfiguration;
+import uk.gov.hmcts.cmc.claimstore.models.idam.User;
+import uk.gov.hmcts.cmc.claimstore.requests.idam.IdamApi;
+import uk.gov.hmcts.cmc.claimstore.services.UserService;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.CoreCaseDataService;
+import uk.gov.hmcts.cmc.domain.models.ClaimState;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@AllArgsConstructor
+@Service
+@Slf4j
+public class TransferCaseStayedService {
+
+    private final CoreCaseDataService coreCaseDataService;
+    private final UserService userService;
+    private final IdamApi idamApi;
+    private final CaseStayedIncrementConfiguration caseStayedIncrementConfiguration;
+
+    public void findCasesForTransfer() {
+
+        User user = userService.authenticateAnonymousCaseWorker();
+        String authorisation = user.getAuthorisation();
+        String userId = idamApi.retrieveUserDetails(authorisation).getId();
+
+        Integer pageNumber = caseStayedIncrementConfiguration.getPageIncrement();
+
+        log.info("Comparing cases to update into ccd");
+
+        if (pageNumber < getNumberOfPages(authorisation, userId)) {
+            compareCases(authorisation, userId, pageNumber);
+            caseStayedIncrementConfiguration.setPageIncrement(pageNumber + 1);
+        } else if (pageNumber.equals(getNumberOfPages(authorisation, userId))) {
+            compareCases(authorisation, userId, pageNumber);
+            caseStayedIncrementConfiguration.setPageIncrement(1);
+        }
+    }
+
+    public void compareCases(String authorisation, String userId, Integer pageNumber) {
+        Integer numberOfPages = getNumberOfPages(authorisation, userId);
+
+        var listOfCases = listCasesWithDeadLIne(
+            authorisation,
+            userId,
+            pageNumber <= numberOfPages && pageNumber > 0
+                ? pageNumber : 1
+        );
+
+        LocalDate currentDate = LocalDate.now();
+
+        JSONArray listOfCasesJson = !listOfCases.isEmpty()
+            ? new JSONArray(listOfCases) : null;
+
+        for (int caseIndex = 0; caseIndex < listOfCases.size(); caseIndex++) {
+            String intentionToProceedDeadline = null;
+            Long caseId = null;
+
+            if (listOfCasesJson.get(caseIndex) != null
+                && !listOfCasesJson.isEmpty()
+                && listOfCasesJson != null
+            ) {
+
+                intentionToProceedDeadline = listOfCasesJson
+                    .getJSONObject(caseIndex)
+                    .get("intentionToProceedDeadline").toString();
+
+                caseId = Long.parseLong(
+                    listOfCasesJson
+                        .getJSONObject(caseIndex)
+                        .get("id").toString());
+            }
+
+            boolean currentDateAfter = intentionToProceedDeadline != null && currentDate
```

Comment:
How about adding this to `searchCriteria` so we get the filtered list from CCD to process.

Reply:
@sabahirfan thanks for suggestion. I think adding a time window in the `searchCriteria` using the `create_date` would be advantageous, the `intentionToProceedDeadline` can be implied, looking at case data specific results would require looking into the data itself which would undo any efforts of improvements. This also poses the question on how CCD is searching its own data.

I did not find any docs to help in implementing something like `create_date`. please see here: [https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1487512777#HLDForGlobalSearchVersion1-3.2.4CCDGlobalSearchAPI](url).

Please feel free to suggest or if I missed anything.

Reply:
You can add search filter based on `create_date` It would be something like below.
```
public List<CaseDetails> fetchAllForDay(String authorisation, String userId, String day) {
        SearchSourceBuilder searchBuilder = new SearchSourceBuilder();
        searchBuilder.size(1);
        searchBuilder.query(QueryBuilders.boolQuery().must(matchQuery(
            "created_date", day)));

        SearchResult searchResult = coreCaseDataApi.searchCases(authorisation, authTokenGenerator.generate(), "Benefit", searchBuilder.toString());
        int total = searchResult.getTotal();
        log.info("Total for " + day + " is " + total);
        searchBuilder.from(pagesize);
        int numberOfPages = total/pagesize;

        List<CaseDetails> caseDetails = IntStream.rangeClosed(0, numberOfPages).boxed()
            .flatMap(pageNumber -> fetchPageEs(authorisation, userId, pageNumber, day).stream())
            .collect(Collectors.toList());
        return caseDetails;
    }
```

Reply:
Do you have a link to the codebase for this?